### ISSUE TITLE
Proposal: broadcast events based on the methods called on navigator

### ISSUE THREAD

**Issue by ericraio** [react-native-simple-router-community/react-native-simple-router#15]
Currently, I have a redux/react application. I would love to be able to respond to events whenever the router has pushed or popped an item on the route stack.

if this sounds acceptable, it's very easy to implement and i can submit a pull request.

**Comment by davidLeonardi** [comment]
I think this is acceptabe.
For the sake of consistency please implement this using an emitter as i did
with the route change

**Comment by ericraio** [comment]
Yeah, that's what I was thinking, i'll submit a PR soon.

**Comment by ericraio** [comment]
@sethx @charpeni 
Hey guys, just catching up on the updates from https://github.com/react-native-simple-router-community/react-native-simple-router/pull/17

 I like the idea better of using RN event emitter and replacing a dependency.

Yeah, I was wondering how to tap into navigators swipe to go back.

What's really nice about this approach is that it gives the developer the ability to add a ref to the router itself and respond to all push or pop events.

For people who use redux, they can maintain the state whenever a route changes if they chose to

``` javascript
componentDidMount() {
  this.refs.router.emitter.addListener("push", (route) => {
    // perform an operation for every push
  })
}

render() {
  return (
    <Router
      ref={"router"}
      ...
    />
  );
}
```

**Comment by davidLeonardi** [comment]
Looks good to me.
My thoughts were the same, so the router could be the central point where devs would attatch their event handlers relative to all things navigation to, kind of like a facade.

Who gets to implement this?

**Comment by ericraio** [comment]
@sethx haha, I can research how to implement this, thought you had a design in mind :dart:

**Comment by davidLeonardi** [comment]
I've been looking at this. Unfortunately pop/push and such do not emit events at a navigator level, and as such there is nothing to attach to.
Another solution is to use AOP and to use advice to have before/after such events.

I'll implement this and lets see how it behaves :)

I've been looking at https://www.npmjs.com/package/aspect-js to add support for this, does anyone have any opinion on it?

**Comment by davidLeonardi** [comment]
Implemented this in #41
